src/aac/plugins/gen_json/gen_json_impl.py

```diff
@@ -10,22 +11,51 @@
 )
 from aac.validator import validation
 
+
 plugin_name = "gen-json"
 
 
-def print_json(architecture_files: list[str]) -> PluginExecutionResult:
-    """Print the parsed_models from the parsed architecture_files values in JSON format."""
+def print_json(architecture_files: list[str], output_directory: str = None) -> PluginExecutionResult:
+    """Prints the parsed_models from the parsed architecture_files values in JSON format.
+
+       (optional) if output directory is given to the cli, parsed_models with be writen to
+        a new file in user defined directory
+
+    """
+
+    def to_json(architecture_file: str, output_directory: str) -> str:
+        file_name = os.path.basename(architecture_file)
+        file_ext = os.path.splitext(file_name)[1]
 
-    def to_json(architecture_file: str) -> str:
         with validation(parse_file, architecture_file) as result:
-            return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
+
+            if output_directory is None:
+                return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
+
+            else:
+                if file_ext == '':
+                    file_name = (file_name + '.json')
+                else:
+                    file_name = file_name.replace(file_ext, '.json')
```

Comment:
How do you set the conditions of the if statement like in line 37 in this new code?

Reply:
His suggestion is leveraging a [ternary operator](https://book.pythontips.com/en/latest/ternary_operators.html#ternary-operators) and ["truthy"](https://www.geeksforgeeks.org/truthy-vs-falsy-values-in-python/) values. So, `file_name.replace(file_ext, '.json')` is the block that executes if `file_ext` is "truthy" (not an empty string), otherwise filename is set to `f'{file_name}.json'`.